docs/proposals/innerloop-to-outerloop.md

```diff
@@ -0,0 +1,138 @@
+# Switching from inner-loop to outer-loop
+
+## Definitions
+
+
+
+### Inner loop
+The Inner loop stage is the development stage where the developer creates an application and it goes through the interation of coding, debugging, testing, locally on the workstation.
+
+### Outer loop validation
+The Outer loop validation stage is the development stage where the developer creates a platform-native build artifact ready to be deployed and distributed on Kubernetes. In case of the Kubernetes platform, a developer would build one or more images from source and deploy them as a Kubernetes Deployment, OpenShift DeploymentConfig or a Knative Service.
+
+### Outer loop Gitops
+The outer loop Gitops stage is the development stage where the developer has validated outerloop by having a built image deployed, and now wishes to deploy the app across multiple environments/stages using CI/CD
+
+
+## End-to-end user experience
+
+
+> As a user of a tool which understands devfiles, 
+I wish to build one or more images from my source code and deploy them on a Kubernetes cluster using the outer-loop guidance from a relevant devfile.
+
+
+The user validates her code in the hot-reload mode in the cluster using: 
+
+```
+odo push
+```
+
+The user validates the Kubernetes-style image-based deployment 
+
+```
+odo deploy
+```
+
+The user chooses to generate deployment manifests, which she would be using to manage Git-driven deployments across multiple stages using CI/CD pipelines
+
+
+```
+odo pipelines bootstrap
```

Comment:
Based on description above, should this be called `odo gitops bootstrap`?

Reply:
Not entirely sure if it would make sense, but would it be possible to break this command into two parts?
One that generates generic stuff that is not application-specific like ReleBindings TektonTemplates etc..
The other that generates manifests that are specific for a given application (devfile).